src/state/machines/turn.ts

```diff
@@ -331,12 +335,26 @@ export const TurnMachine = setup({
           draft.policyMarket.active.push(policyCard);
         }
 
+        const strictProtectionCard = context.policyMarket.acquired.find(
+          (card) => card.name === StrictProtection.cardName,
+        );
+
+        const allowProtection =
+          strictProtectionCard &&
+          draft.policyMarket.funding.length > 0 &&
+          [OilSpill.cardName, HazardousIndustrialSubstances.cardName, Overfishing.cardName].includes(policyCard.name);
+
         draft.policyMarket.deck = without(context.policyMarket.deck, policyCard);
 
         draft.stage = {
-          eventType: policyCard.effect === "implementation" ? "policy_fundingIncrease" : "policy_policyDraw",
+          eventType:
+            policyCard.effect === "implementation"
+              ? "policy_fundingIncrease"
+              : allowProtection
+                ? "policy_allowProtectionActivation"
+                : "policy_policyDraw",
           cause: undefined,
-          effect: [policyCard.uid],
+          effect: allowProtection ? [policyCard.uid, strictProtectionCard.uid] : [policyCard.uid],
```

Comment:
ugh... really don't like exp. pack code leaking into core. Can't we have this code inside OilSpill/Overfishing/... cards?

Maybe use shared guard + action for that?

Reply:
Ok, now there is no more expansion logic in the main machine, let me know what you think. I added a new file for shared expansion actions, this allows us to use the same action logic in all three expansion cards. The flow now goes as follows:

Unlock card stage -> Activate strict protection stage -> Species protected stage -> Done